test/e2e/framework/util.go

```diff
@@ -2876,29 +2877,32 @@ func UnblockNetwork(from string, to string) {
 	}
 }
 
-// PingCommand is the type to hold ping command.
-type PingCommand string
-
-const (
-	// IPv4PingCommand is a ping command for IPv4.
-	IPv4PingCommand PingCommand = "ping"
-	// IPv6PingCommand is a ping command for IPv6.
-	IPv6PingCommand PingCommand = "ping6"
-)
-
 // CheckConnectivityToHost launches a pod to test connectivity to the specified
 // host. An error will be returned if the host is not reachable from the pod.
 //
 // An empty nodeName will use the schedule to choose where the pod is executed.
-func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingCmd PingCommand, timeout int) error {
+func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, port, timeout int) error {
 	contName := fmt.Sprintf("%s-container", podName)
 
-	command := []string{
-		string(pingCmd),
-		"-c", "3", // send 3 pings
-		"-W", "2", // wait at most 2 seconds for a reply
-		"-w", strconv.Itoa(timeout),
-		host,
+	var command []string
+	if TestContext.IPFamily == "ipv4" {
```

Comment:
I think that we don´t need to differentiate here, nc seems to work for both families, no need to specify -6 as a parameter, at least in my tests

```
/ # nc -vz -w 2 2001:db8:1::1 22
Connection to 2001:db8:1::1 22 port [tcp/ssh] succeeded!
/ # nc -vz -w 2 172.17.0.1 22
Connection to 172.17.0.1 22 port [tcp/ssh] succeeded!
```

However, if you still want to differentiate I suggest to not rely in the TestContext.IPFamily here, and use the `host string` to detect the IP family, i.e using this function https://github.com/kubernetes/utils/blob/6c36bc71fc4aeb1f49801054e71aebdaef1fbeb4/net/net.go#L119

Reply:
Thanks! I already combined the two commands together.